apollo-api/src/main/java/com/apollographql/apollo/api/Operation.java

```diff
@@ -100,6 +103,24 @@ protected Variables() {
         }
       };
     }
+
+    /**
+     * Serializes variables as JSON string to be sent to the GraphQL server.
+     *
+     * @param scalarTypeAdapters adapters for custom GraphQL scalar types
+     * @return JSON string
+     * @throws IOException
+     */
+    public final String marshal(@NotNull final ScalarTypeAdapters scalarTypeAdapters) throws IOException {
```

Comment:
Would it make sense to have an overloaded function without the scalarTypeAdapters? It would make it easier for those who don't have anything custom.

Reply:
If we do this for `Variables.marshal`, we should also certainly do it for `Operation.parse()` ? Another option would be to add a SingleTon empty scalaraTypeAdapters ?  

```  public static ScalarTypeAdapters Empty = new ScalarTypeAdapters(new HashMap<ScalarType, CustomTypeAdapter>())
```

Reply:
That's a good idea. My comment was also just a simple idea and not a blocker for this PR. 

I actually believe it would be better as a follow-up.

Reply:
sure makes sense, will address this in next PR.

Reply:
Added new API:
```
    public final String marshal() throws IOException {
      return marshal(ScalarTypeAdapters.DEFAULT);
    }
```

For parse it will be in the next PR, as it will involve to update code generation fixtures.